app/models/manageiq/providers/openshift/infra_manager.rb

```diff
@@ -0,0 +1,74 @@
+ManageIQ::Providers::Kubevirt::InfraManager.include(ActsAsStiLeafClass)
+
+class ManageIQ::Providers::OpenShift::InfraManager < ManageIQ::Providers::Kubevirt::InfraManager
+  belongs_to :parent_manager,
+             :foreign_key => :parent_ems_id,
+             :class_name  => "ManageIQ::Providers::ContainerManager"
+
+  delegate :authentication_check,
+           :authentication_for_summary,
+           :authentication_token,
+           :authentications,
+           :endpoints,
+           :default_endpoint,
+           :zone,
+           :to        => :parent_manager,
+           :allow_nil => true
+
+  #
+  # This is the list of features that this provider supports:
+  #
+  supports :catalog
+  supports :provisioning
+
+  def self.ems_type
+    @ems_type ||= "openshift".freeze
```

Comment:
I think this has to be different than the container manager, so maybe something like `openshift_virt` or `openshift_infra`?

Reply:
:+1: openshift_infra would be good (OpenStack undercloud/InfraManager is openstack_infra)